src/com/tony/server/WorkerThread.java

```diff
@@ -8,19 +8,28 @@
 public class WorkerThread implements Runnable {
 
     private final ResponseDeterminer responseDeterminer;
+    private final Logger logger;
     private Socket clientSocket;
 
-    public WorkerThread(Socket clientSocket, ResponseDeterminer responseDeterminer) throws IOException {
+    public WorkerThread(Socket clientSocket, ResponseDeterminer responseDeterminer, Logger logger) throws IOException {
         this.clientSocket = clientSocket;
         this.responseDeterminer = responseDeterminer;
+        this.logger = logger;
     }
 
     @Override
     public void run() {
         try {
             Request request = RequestParser.parseRequest(getInputStream());
+            logger.storeLog("Received Request:\n");
+            logger.storeLog(request.toString());
+
             Response response = responseDeterminer.determineResponse(request);
-            getOutputStream().write(response.respond());
+            byte[] responseBytes = response.respond();
+            logger.storeLog("Sending Response:\n");
```

Comment:
You have a new line appended to your messages inside of `#storeLog`, do you actually want new lines after each log message?